File: xstudyout/src/main/java/br/com/hackatur/xtudyout/config/FirebaseConfig.java
package br.com.hackatur.xtudyout.config;

import br.com.hackatur.xtudyout.utils.FileHelper;
import com.google.auth.oauth2.GoogleCredentials;
import com.google.cloud.firestore.Firestore;
import com.google.cloud.firestore.FirestoreOptions;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.io.IOException;

@Configuration
public class FirebaseConfig {

    @Bean
    public Firestore initFirebase() throws IOException {

        GoogleCredentials google = GoogleCredentials.fromStream(FileHelper.getFireBaseCredentialStream());

        FirestoreOptions firestoreOptions =
                FirestoreOptions.getDefaultInstance().toBuilder()
                        .setCredentials(google)
                        .build();

        return firestoreOptions.getService();

    }


}

File: xstudyout/src/main/kotlin/br/com/hacktur/xstudyout/config/FirebaseConfig.kt
import com.google.auth.oauth2.GoogleCredentials
import com.google.cloud.firestore.Firestore
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import com.google.firebase.cloud.FirestoreClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class FirebaseConfig {

    @Bean
    fun initFirebase(): Firestore {
        val firebaseDatabseUrl = "https://exchange-c9b48.firebaseio.com/"
        val google: GoogleCredentials = GoogleCredentials.fromStream(FileHelper().getFireBaseCredentialStream().byteInputStream())

        val options = FirebaseOptions.Builder()
                .setCredentials(google)
                .setDatabaseUrl(firebaseDatabseUrl)
                .setDatabaseAuthVariableOverride(null)
                .build()


        FirebaseApp.initializeApp(options)

        return FirestoreClient.getFirestore()
    }
}